src/main/java/seedu/address/model/event/IsolatedEvent.java

```diff
@@ -1,15 +1,48 @@
 package seedu.address.model.event;
 
 import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 
 /**
  * Represents an {@code Event} that happens only once.
  */
 public class IsolatedEvent extends Event implements Comparable<IsolatedEvent> {
 
-    // TODO: Add String representation of IsolatedEvent
-    public IsolatedEvent(String eventName) {
+    public static final String VALIDATION_REGEX_EVENTNAME = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String MESSAGE_CONSTRAINTS_EVENTNAME =
+            "Event name should only contain alphanumeric characters and spaces, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS_DATE =
+            "Date should be in the format: dd/MM/yyyy HH:mm";
+
+    private final String eventName;
+    private final LocalDateTime startDate;
+    private final LocalDateTime endDate;
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+
+    /**
+     * Constructor for IsolatedEvent class.
+     * @param eventName
+     * @param startDate
+     * @param endDate
```

Comment:
Perhaps this Javadoc could also explain the parameters, return values and exceptions